Gems/Terrain/Code/Source/TerrainRenderer/TerrainMacroMaterialManager.cpp

```diff
@@ -256,16 +256,18 @@ namespace Terrain
         UpdateImageIndex(shaderData.m_colorMapId, macroMaterialData.m_colorImage);
         UpdateImageIndex(shaderData.m_normalMapId, macroMaterialData.m_normalImage);
 
-        MacroMaterialMetaData& metaData = m_materialData.GetElement<1>(materialHandle.GetIndex());
-        metaData.m_priority = macroMaterialData.m_priority;
+        MacroMaterialPriority& priority = m_materialData.GetElement<1>(materialHandle.GetIndex());
+        priority.m_priority = macroMaterialData.m_priority;
+        priority.m_hash = uint32_t(AZ::u64(macroMaterialData.m_entityId) >> 32) ^ uint32_t(AZ::u64(macroMaterialData.m_entityId) & 0xFFFFFFFF);
```

Comment:
The higher 32 bits are the count of entities. For the purpose here, I think we can consider using the lower 32 bits directly, or the higher 32 bits as the count never decreases.
See in `EntityId Entity::MakeId()` in `Entity.cpp`.

Reply:
Looks like the lower 32 bits is only based on the process id / start time / etc and would likely be the same for lots of entity ids. It'd probably be safe for now to use only the higher 32 bits, but I still think it's best to use both to be on the safe side - you never know when how the IDs are generated under the hood will change, but it's likely they'll stay 64 bit.

Reply:
You _could_ also use the lower 32 bits of the EntityId hash i think:
`priority.m_priority = static_cast<uint32_t>(AZStd::hash<AZ::EntityId>(macroMaterialData.m_entityId));`

Reply:
@mbalfour-amzn I don't think that would work - the EntityId hash just does
```
AZStd::hash<AZ::u64> hasher;
return hasher(static_cast<AZ::u64>(id));
```
And hasher under the hood is just
```
constexpr result_type operator()(const argument_type& value) const
{
    return static_cast<result_type>(value);
}
```
So it looks like it doesn't transform the value at all - which kinda makes sense - the entity Id is just a 64 bit number under the hood, and if you need a 64 bit value for a hash, just using the number as-is should work as well as anything.